src/renderer/shared/api/xcm/xcmService.ts

```diff
@@ -143,94 +181,115 @@ const getRelativeAssetLocation = (
+          },
+        },
+        fun: {
+          Fungible: amount.toNumber(),
         },
       },
-      fun: {
-        Fungible: amount.toNumber(),
-      },
-    },
-  });
+    ],
+  };
 };
 
 const getAbsoluteAssetLocation = (
   api: ApiPromise,
   amount: BN,
   assetLocation?: LocalMultiLocation,
-): VersionedMultiAsset | undefined => {
+): Object | undefined => {
   if (!assetLocation) return;
 
-  return api.createType('VersionedMultiAsset', {
-    V2: {
-      id: {
-        Concrete: {
-          parents: 1,
-          interior: Object.values(assetLocation).length ? createJunctionFromObject(assetLocation) : 'Here',
+  return {
+    V2: [
+      {
+        id: {
+          Concrete: {
+            parents: 1,
+            interior: Object.values(assetLocation).length ? createJunctionFromObject(assetLocation) : 'Here',
+          },
+        },
+        fun: {
+          Fungible: amount.toNumber(),
         },
       },
-      fungibility: {
-        Fungible: amount.toNumber(),
-      },
-    },
-  });
+    ],
+  };
 };
 
 const getConcreteAssetLocation = (
   api: ApiPromise,
   amount: BN,
   assetLocation?: LocalMultiLocation,
-): VersionedMultiAsset | undefined => {
+): Object | undefined => {
   if (!assetLocation) return;
 
   const { parents, ...location } = assetLocation;
 
-  return api.createType('VersionedMultiAsset', {
-    V2: {
-      id: {
-        Concrete: {
-          parents,
-          interior: Object.values(location).length ? createJunctionFromObject(location) : 'Here',
+  return {
+    V2: [
+      {
+        id: {
+          Concrete: {
+            parents,
+            interior: Object.values(location).length ? createJunctionFromObject(location) : 'Here',
+          },
+        },
+        fun: {
+          Fungible: amount.toNumber(),
         },
       },
-      fun: {
-        Fungible: amount.toNumber(),
-      },
-    },
-  });
+    ],
+  };
 };
 
 export const getDestinationLocation = (
-  api: ApiPromise,
   originChain: Pick<Chain, 'parentId'>,
   destinationParaId?: number,
   accountId?: AccountId,
-): VersionedMultiLocation | undefined => {
+): Object | undefined => {
   if (originChain.parentId && destinationParaId) {
-    return getSiblingLocation(api, destinationParaId, accountId);
+    return getSiblingLocation(destinationParaId, accountId);
   }
 
   if (originChain.parentId) {
-    return getParentLocation(api, accountId);
+    return getParentLocation(accountId);
   }
 
   if (destinationParaId) {
-    return getChildLocation(api, destinationParaId, accountId);
+    return getChildLocation(destinationParaId, accountId);
   }
 
   return undefined;
 };
 
-const getChildLocation = (api: ApiPromise, parachainId: number, accountId?: AccountId): VersionedMultiLocation => {
+export const getAccountLocation = (accountId?: AccountId): Object | undefined => {
+  return {
+    V2: {
+      parents: 0,
+      interior: {
+        X1: {
+          accountId32: {
```

Comment:
Soon we will be integrating EVM chains will use AccountId20, dunno if something should be changed right now, just for you to have a context